controller/src/main/java/org/apache/rocketmq/controller/impl/manager/ReplicasInfoManager.java

```diff
@@ -160,25 +161,30 @@ public ControllerResult<ElectMasterResponseHeader> electMaster(
             // First, check whether the master is still active
             final String oldMaster = syncStateInfo.getMasterAddress();
             if (StringUtils.isNoneEmpty(oldMaster) && brokerAlivePredicate.test(brokerInfo.getClusterName(), oldMaster)) {
-                String err = String.format("The old master %s is still alive, not need to elect new master for broker %s", oldMaster, brokerInfo.getBrokerName());
-                log.warn("{}", err);
-                result.setCodeAndRemark(ResponseCode.CONTROLLER_INVALID_REQUEST, err);
-                return result;
+
+                if (StringUtils.isBlank(assignBrokerAddress) || StringUtils.equals(oldMaster, assignBrokerAddress)) {
+                    String err = String.format("The old master %s is still alive, not need to elect new master for broker %s", oldMaster, brokerInfo.getBrokerName());
+                    log.warn("{}", err);
+                    result.setCodeAndRemark(ResponseCode.CONTROLLER_INVALID_REQUEST, err);
+                    return result;
+                }
             }
 
             // Try elect a master in syncStateSet
             if (syncStateSet.size() > 1) {
-                boolean electSuccess = tryElectMaster(result, brokerName, syncStateSet, candidate ->
-                    !candidate.equals(syncStateInfo.getMasterAddress()) && brokerAlivePredicate.test(brokerInfo.getClusterName(), candidate));
+                boolean electSuccess = tryElectMaster(result, brokerName, assignBrokerAddress, syncStateSet, candidate ->
+                    (StringUtils.isEmpty(assignBrokerAddress) ? !StringUtils.equals(candidate, syncStateInfo.getMasterAddress()) : true)
```

Comment:
This judgment seems unnecessary. -> StringUtils.isEmpty(assignBrokerAddress)

Reply:
sorry, I don't understant, what's the meaning of this line?

Reply:
Redundant check, I remove it